aws-iot-softwarepackageversion/src/main/java/software/amazon/iot/softwarepackageversion/ReadHandler.java

```diff
@@ -0,0 +1,93 @@
+
+        return proxy.initiate(CALL_GRAPH, proxyClient, resourceModel, callbackContext)
+                .translateToServiceRequest(Translator::translateToReadRequest)
+                .makeServiceCall(this::readResource)
+                .done((getPackageVersionRequest, getPackageVersionResponse, sdkProxyClient, model, context) ->
+                        constructSuccessfulProgressEventFromResponse(sdkProxyClient, getPackageVersionResponse));
+    }
+
+    /**
+     * Implement client invocation of the read request through the proxyClient, which is already initialised with
+     * caller credentials, correct region and retry settings
+     * @param getPackageVersionRequest the aws service request to describe a resource
+     * @param proxyClient the aws service client to make the call
+     * @return describe resource response
+     */
+    private GetPackageVersionResponse readResource(
+            GetPackageVersionRequest getPackageVersionRequest,
+            ProxyClient<IotClient> proxyClient) {
+        try {
+            GetPackageVersionResponse getPackageVersionResponse = proxyClient.injectCredentialsAndInvokeV2(
+                    getPackageVersionRequest, proxyClient.client()::getPackageVersion);
+            logger.log(String.format("%s [%s, %s] has successfully been read.",
+                    ResourceModel.TYPE_NAME, getPackageVersionRequest.packageName(), getPackageVersionRequest.versionName()));
+            return getPackageVersionResponse;
+        } catch (final IotException e) {
+            throw Translator.translateIotExceptionToHandlerException(getPackageVersionRequest.packageName() + ":" + getPackageVersionRequest.versionName(), OPERATION, e);
+        }
+    }
+
+    /**
+     * Implement client invocation of the read request through the proxyClient, which is already initialised with
+     * caller credentials, correct region and retry settings
+     * @param getPackageVersionResponse the aws service describe resource response
+     * @return progressEvent indicating success, in progress with delay callback or failed state
+     */
+    private ProgressEvent<ResourceModel, CallbackContext> constructSuccessfulProgressEventFromResponse(
+            ProxyClient<IotClient> proxyClient,
+            GetPackageVersionResponse getPackageVersionResponse) {
+        final ResourceModel resourceModel = Translator.translateFromReadResponse(getPackageVersionResponse);
+
+        try {
+            List<Tag> tags = listTags(proxyClient, getPackageVersionResponse.packageVersionArn());
+            resourceModel.setTags(Tagging.translateSdkTagsToMap(tags));
+        } catch (final IotException e) {
+            if (e.statusCode() != HttpStatusCode.FORBIDDEN) {
+                throw Translator.translateIotExceptionToHandlerException(
+                        getPackageVersionResponse.packageName() + ":" + getPackageVersionResponse.versionName(), OPERATION, e);
+            }
```

Comment:
Add an else statement here catch other exceptions other than forbidden

Reply:
If I remove it, it fails the cfn contract test case. I guess that is why this same logic is added in the other resource handlers as well.